src/API.jl

```diff
@@ -1278,7 +1278,7 @@ function precompile(ctx::Context; internal_call::Bool=false, kwargs...)
             err_str = ""
             n_direct_errs = 0
             for (dep, err) in failed_deps
-                if dep in direct_deps
+                if fail_indirect_deps || (dep in direct_deps)
                     err_str *= "\n" * "$dep" * "\n\n" * err * (n_direct_errs > 0 ? "\n" : "")
                     n_direct_errs += 1
```

Comment:
Looks like the `pkgerror` later needs to be revised as it assumes direct dependencies